### PR TITLE
Top CLI commands

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/2157

I replaced our extraneous "Featured docs" boxes with a "Core commands" section. Tried to keep the copy light because we already have so much in the boxes aboce.

### CODE REVIEW

File: astro/cli/reference.md [hunk 1]
```diff
@@ -17,18 +17,13 @@ The Astronomer product you're using determines the behavior and format of comman
 
 ## Core Commands
```
[the remark this answers is not in the document]
Comment (lzdanski) [reply astro/cli/reference.md:18]: FWIW - when i was going through the getting started and quickstarts, the internal search function for the CLI commands is really good. It was easy to find the CLI ref pages vs in context use in our Docs. So having the lists identical is probably okay?

**Review by lzdanski — APPROVED** [review]

File: astro/cli/reference.md [hunk 2]
```diff
@@ -17,18 +17,13 @@ The Astronomer product you're using determines the behavior and format of comman
...
 
-We expect that you'll use these commands most often when managing your Astro projects and Deployments:
-
 - [`astro login`](cli/astro-login.md)
 - [`astro dev init`](cli/astro-dev-init.md)
 - [`astro dev start`](cli/astro-dev-start.md)
-- [`astro dev stop`](cli/astro-dev-stop.md)
 - [`astro dev restart`](cli/astro-dev-restart.md)
 - [`astro deploy`](cli/astro-deploy.md)
 - [`astro run`](cli/astro-run.md)
```
Comment (jwitz) [inline astro/cli/reference.md:25]:
```suggestion
```

**Review by paolaperaza — COMMENTED** [review]
@jwitz Thanks for getting this together. Here's my initial thoughts:

- I think we should bump up the `Features` section, remove the boxes, and turn it into a bullet list that starts with, "With the Astro CLI, you can..."
- Then have the core commands section right after it
- That's it

Re: Viraj's feedback, down the line we can consider:
- A local-only quickstart in Astro CLI tab that's different than Astro quickstart + tutorial
- Bringing over more local troubleshooting content from "develop a project" (e.g. docker compose)

WDYT about making those initial changes in this PR?

**Comment by jwitz** [comment]
@paolaperaza I made those initial changes, so this is good for another review!

**Pull request merged**